src/SFA.DAS.Comitments.AcademicYearEndProcessor.WebJob/Updater/AcademicYearEndExpiryProcessor.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using SFA.DAS.Commitments.Domain.Data;
+using SFA.DAS.Commitments.Domain.Entities.DataLock;
+using SFA.DAS.Commitments.Domain.Interfaces;
+
+namespace SFA.DAS.Comitments.AcademicYearEndProcessor.WebJob.Updater
+{
+    public class AcademicYearEndExpiryProcessor : IAcademicYearEndExpiryProcessor
+    {
+        private readonly IAcademicYearDateProvider _academicYearProvider;
+        private readonly IDataLockRepository _dataLockRepository;
+        private readonly DataLockErrorCode _expirableErrorCodes;
+        private readonly ICurrentDateTime _currentDateTime;
+        private List<DataLockStatus> _expirableDatalocks = new List<DataLockStatus>();
+
+        public AcademicYearEndExpiryProcessor(IAcademicYearDateProvider academicYearProvider,
+            IDataLockRepository dataLockRepository, DataLockErrorCode expirableErrorCodes,
+            ICurrentDateTime currentDateTime)
+        {
+            _dataLockRepository = dataLockRepository ?? throw new ArgumentException("dataLockRepository");
+            _currentDateTime = currentDateTime ?? throw new ArgumentException("currentDateTime");
+            _academicYearProvider = academicYearProvider ?? throw new ArgumentException("academicYearProvider");
+            _expirableErrorCodes = expirableErrorCodes;
+        }
+
+        public async Task RunUpdate()
+        {
+
+            if (_currentDateTime.Now > _academicYearProvider.CurrentAcademicYearEndDate ||
+                _currentDateTime.Now < _academicYearProvider.CurrentAcademicYearStartDate)
+            {
+                throw new InvalidAcademicYearException($"The current time {_currentDateTime.Now} does not fall within the specified Academic Year: {_academicYearProvider.CurrentAcademicYearStartDate}-{_academicYearProvider.CurrentAcademicYearEndDate}.");
+            }
+
+            if (
+                _currentDateTime.Now >= _academicYearProvider.CurrentAcademicYearStartDate
+                && _currentDateTime.Now < _academicYearProvider.LastAcademicYearFundingPeriod)
+            {
+                return;
```

Comment:
This is worth logging - we would't DevOps to run it a few seconds before the time and think everything is ok.